app/models/edition.rb

```diff
@@ -681,11 +684,15 @@ def has_policy_areas?
   end
 
   def locked?
-    document.locked?
+    document&.locked?
```

Comment:
This breaks several test cases that create editions without documents, I thought it a safe shorthand response to this that didn't involve editing cases to include information they didn't explicitly need.

Reply:
Just putting it down so it's on the record - there is some question over how Whitehall requires documents - especially as the test factory doesn't actually add them without a trait - so for now this is going up as is, and a change should come later.

Reply:
Is that a change being made in this PR or a separate PR?  I feel a bit uneasy about making a change to code just to make a test pass rather than configuring tests to work with the code.  The tests tell us that a document is required to create an edition: https://github.com/alphagov/whitehall/blob/master/test/unit/edition/validation_test.rb#L21-L22

Reply:
I also feel uneasy about this. Intent is to make a separate PR but I've not gotten around to it yet (meetings, issues with local environment). If you'd rather it'd be in this PR I can solve it but it will delay it until tomorrow.

Reply:
I looked into this with Jon for a bit. It's a bit of a confusing beast - edition doesn't require a document at a db level (which may just be a mistake) but one is always created as part of the [`before_validation`](https://github.com/alphagov/whitehall/blob/7b5c5a086b89cb62ffba62b152a0a8dcfc10c8e6/app/models/edition/identifiable.rb#L7) callback and [some](https://github.com/alphagov/whitehall/blob/7b5c5a086b89cb62ffba62b152a0a8dcfc10c8e6/app/models/edition/identifiable.rb#L8) ([but not all](https://github.com/alphagov/whitehall/blob/7b5c5a086b89cb62ffba62b152a0a8dcfc10c8e6/app/models/edition/identifiable.rb#L9)) callback methods expect a document to exist.

What I guess makes this situation here a bit confusing is for this to be a problem the `before_validation` callback needs to have not run and the `before_save` one has to have run. I ran a [quick isolated version](https://ci.integration.publishing.service.gov.uk/job/whitehall/job/locked-edition/1/console) and it looks like we only have the one failing test for this: https://github.com/alphagov/whitehall/blob/e34f3c1b263201838394375b600952a9ad9fc56b/test/unit/edition/images_test.rb#L82-L104

I think we could resolve this by putting a document in the [hash of attributes](https://github.com/alphagov/whitehall/blob/e34f3c1b263201838394375b600952a9ad9fc56b/test/unit/edition/images_test.rb#L109-L119) or in the [valid_edition_attributes](https://github.com/alphagov/whitehall/blob/e34f3c1b263201838394375b600952a9ad9fc56b/test/unit/edition/images_test.rb#L142-L149) method.